src/main/java/frc/robot/sensors/IrSensor.java

```diff
@@ -0,0 +1,71 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2018-2019 FIRST. All Rights Reserved.                        */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+/**
+ * Code by Charlie Macdonald.
+ * Last modified on Feb 6th, 2020.
+ */
+
+package frc.robot.sensors;
+
+import edu.wpi.first.wpilibj.AnalogInput;
+import edu.wpi.first.wpilibj.MedianFilter;
+import edu.wpi.first.wpilibj.Sendable;
+import edu.wpi.first.wpilibj.smartdashboard.SendableBuilder;
+import edu.wpi.first.wpilibj.smartdashboard.SendableRegistry;
+
+public class IrSensor implements Sendable {
+    private static final int kIRPort = 1; //set IR sensor port
+        //TODO: The IR sensor's port may change, depending on the robot setup. (Use a robot-specific config to set this?)
+
+    //median filter to discard outliers. Filters over 10 samples
+    private final MedianFilter m_filter = new MedianFilter(10);
+
+    private final AnalogInput m_irsensor;
+
+    //Set the formula values to calculate IR sensor distance:
+        //Values for short range (Sharp GP2Y0A41SK0F) 4-30cm sensor: 12.8528, -0.007545, -3.92, 30, 4.5.
+        //Note: If a different IR sensor is used, these values will need to be updated.
+    private final double irSlopeValue = 12.8528;
```

Comment:
Since these are constants do you mind CAPITALIZING them?

Reply:
Sure thing. Also, I believe words in constants are separated by underscores, i.e. IR_SLOPE_VALUE, so I'll do that.